solo-io/gloo
# K8s tests in kind

# Description

- Move tests which rely on k8s to the kube2e test package, which uses KinD
  - Remove `ShouldRunKubesTests()` checks
  - Ensure all tests use actual kubes client
  - Add relevant setup/breakdown steps to kube2e's test suite file
  - Include comment referring to package each file is meant to test
  - Cleanup tests, removing code that was tied to the persistent cluster paradigm
- Remove `RUN_KUBES_TESTS` from env.go test file, `Kuberentes()` from requirements.go test file
- Move kubernetes happy_path test to kube2e
- Remove kube client from aws e2e tests which do not appear to require it 

# Context

This is the first step towards migrating off of build bot for testing

We will stop setting `RUN_KUBES_TESTS` in a followup PR, once it's clear that no tests are breaking

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/main/changelogutils) which references the issue that is resolved.

BOT NOTES: 
resolves https://github.com/solo-io/solo-projects/issues/4907
